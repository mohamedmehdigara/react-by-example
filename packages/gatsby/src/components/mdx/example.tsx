import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
