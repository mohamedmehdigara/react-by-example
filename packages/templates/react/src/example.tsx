import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
