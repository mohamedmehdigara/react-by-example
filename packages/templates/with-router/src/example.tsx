import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
