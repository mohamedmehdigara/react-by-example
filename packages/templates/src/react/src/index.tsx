import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
