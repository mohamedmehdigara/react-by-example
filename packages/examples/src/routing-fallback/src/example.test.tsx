import { cleanup, render, RenderResult } from '@testing-library/react'
import React, { FC } from 'react'
import { MemoryRouter } from 'react-router'
import { Example } from './example'

describe('Example', () => {
  let component: RenderResult

  interface TestProps {
    location: string
  }
  const Test: FC<TestProps> = ({ location }) => {
    return (
      <MemoryRouter initialEntries={[location]}>
        <Example />
      </MemoryRouter>
    )
  }

  afterEach(() => {
    cleanup()
  })

  describe('when home', () => {
    beforeEach(() => {
      component = render(<Test location="/" />)
    })

    it('should snapshot nav', () => {
      expect(component.getAllByRole('navigation')).toMatchSnapshot()
    })

    it('should snapshot', () => {
      expect(component.getByRole('article')).toMatchSnapshot()
    })

    it('should set active class', () => {
      expect(component.getByText('home')).toHaveClass('activeLink')
    })
  })

  describe('when pricing', () => {
    beforeEach(() => {
      component = render(<Test location="/pricing" />)
    })

    it('should snapshot', () => {
      expect(component.getByRole('article')).toMatchSnapshot()
    })

    it('should set active class', () => {
      expect(component.getByText('pricing')).toHaveClass('activeLink')
    })
  })

  describe('when about', () => {
    beforeEach(() => {
      component = render(<Test location="/about" />)
    })

    it('should snapshot', () => {
      expect(component.getByRole('article')).toMatchSnapshot()
    })

    it('should set active class', () => {
      expect(component.getByText('about')).toHaveClass('activeLink')
    })
  })

  describe('when 404', () => {
    beforeEach(() => {
      component = render(<Test location="/404" />)
    })

    it('should snapshot', () => {
      expect(component.getByRole('article')).toMatchSnapshot()
    })
  })
})
