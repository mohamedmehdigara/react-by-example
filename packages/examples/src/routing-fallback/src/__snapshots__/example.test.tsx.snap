// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example when 404 should snapshot 1`] = `
<article>
  404 page
</article>
`;

exports[`Example when about should snapshot 1`] = `
<article>
  about page
</article>
`;

exports[`Example when home should snapshot 1`] = `
<article>
  home page
</article>
`;

exports[`Example when home should snapshot nav 1`] = `
Array [
  <nav>
    <a
      aria-current="page"
      class="activeLink"
      href="/"
    >
      home
    </a>
     
    <a
      href="/pricing"
    >
      pricing
    </a>
     
    <a
      href="/about"
    >
      about
    </a>
     
    <a
      href="/this-page-does-not-exist"
    >
      broken link
    </a>
  </nav>,
]
`;

exports[`Example when pricing should snapshot 1`] = `
<article>
  pricing page
</article>
`;
